kernel/build.sbt

```diff
@@ -0,0 +1,201 @@
+    )
+  )
+
+/*
+ ***********************
+ * ScalaStyle settings *
+ ***********************
+ */
+ThisBuild / scalastyleConfig := baseDirectory.value / "scalastyle-config.xml"
+
+// Not used since scala is test-only
+lazy val compileScalastyle = taskKey[Unit]("compileScalastyle")
+lazy val testScalastyle = taskKey[Unit]("testScalastyle")
+
+lazy val scalaStyleSettings = Seq(
+  compileScalastyle := (Compile / scalastyle).toTask("").value,
+  Compile / compile := ((Compile / compile) dependsOn compileScalastyle).value,
+  testScalastyle := (Test / scalastyle).toTask("").value,
+  Test / test := ((Test / test) dependsOn testScalastyle).value
+)
+
+/*
+ ********************
+ * Release settings *
+ ********************
+ */
+
+// Looks some of release settings should be set for the root project as well.
+publishArtifact := false  // Don't release the root project
+publish / skip := true
+publishTo := Some("snapshots" at "https://oss.sonatype.org/content/repositories/snapshots")
+releaseCrossBuild := false
+releaseProcess := Seq[ReleaseStep](
+  checkSnapshotDependencies,
+  inquireVersions,
+  runTest,
+  setReleaseVersion,
+  commitReleaseVersion,
+  tagRelease,
+  releaseStepCommandAndRemaining("+publishSigned"),
+  setNextVersion,
+  commitNextVersion
+)
+
+/**
+ * Release settings for artifact that contains only Java source code
+ */
+lazy val javaOnlyReleaseSettings = Seq(
+  // drop off Scala suffix from artifact names
+  crossPaths := false,
+  // we publish jars for each scalaVersion in crossScalaVersions. however, we only need to publish
+  // one java jar. thus, only do so when the current scala version == default scala version
+  publishArtifact := scalaBinaryVersion.value == "2.12",
+  // exclude scala-library from dependencies in generated pom.xml
+  autoScalaLibrary := false,
+)
+
+lazy val releaseSettings = Seq(
+  publishMavenStyle := true,
+  publishArtifact := true,
+  Test / publishArtifact := false,
+  releasePublishArtifactsAction := PgpKeys.publishSigned.value,
+  releaseCrossBuild := true,
+  pgpPassphrase := sys.env.get("PGP_PASSPHRASE").map(_.toArray),
+  sonatypeProfileName := "io.delta", // sonatype account domain name prefix / group ID
+  credentials += Credentials(
+    "Sonatype Nexus Repository Manager",
+    "oss.sonatype.org",
+    sys.env.getOrElse("SONATYPE_USERNAME", ""),
+    sys.env.getOrElse("SONATYPE_PASSWORD", "")
+  ),
+  publishTo := {
+    val nexus = "https://oss.sonatype.org/"
+    if (isSnapshot.value) {
+      Some("snapshots" at nexus + "content/repositories/snapshots")
+    } else {
+      Some("releases"  at nexus + "service/local/staging/deploy/maven2")
+    }
+  },
+  licenses += ("Apache-2.0", url("http://www.apache.org/licenses/LICENSE-2.0")),
+  pomExtra :=
+    <url>https://github.com/delta-io/connectors</url>
+      <scm>
+        <url>git@github.com:delta-io/connectors.git</url>
+        <connection>scm:git:git@github.com:delta-io/connectors.git</connection>
```

Comment:
since we dont know what the release structure is going to be like.. i am inclined to not make any assumptions here and forget about those later and accidentally release something wrong. I think we should remove this entire release setting stuff. 

all we should ensure that we are generating a pure java artifact when locally published without _2.12 stuff at the end.

Reply:
Removed all the extra settings. Verified that `publishM2` does as desired:
```
sbt:kernel> publishM2
[info] scalastyle using config /Users/allison.portis/m1/delta/kernel/scalastyle-config.xml
[info] scalastyle using config /Users/allison.portis/m1/delta/kernel/scalastyle-config.xml
[info] Wrote /Users/allison.portis/m1/delta/kernel/kernel-api/target/delta-kernel-api-0.1.0-SNAPSHOT.pom
[info] Wrote /Users/allison.portis/m1/delta/kernel/kernel-default/target/delta-kernel-default-0.1.0-SNAPSHOT.pom
[info] scalastyle Processed 0 file(s)
[info] scalastyle Processed 0 file(s)
[info] scalastyle Found 0 errors
[info] scalastyle Found 0 errors
[info] scalastyle Found 0 warnings
[info] scalastyle Found 0 warnings
[info] scalastyle Found 0 infos
[info] scalastyle Found 0 infos
[info] scalastyle Finished in 8 ms
[info] scalastyle Finished in 8 ms
[success] created output: /Users/allison.portis/m1/delta/kernel/kernel-api/target
[success] created output: /Users/allison.portis/m1/delta/kernel/kernel-default/target
[info] 	published delta-kernel-api to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-api/0.1.0-SNAPSHOT/delta-kernel-api-0.1.0-SNAPSHOT.pom
[info] 	published delta-kernel-api to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-api/0.1.0-SNAPSHOT/delta-kernel-api-0.1.0-SNAPSHOT.jar
[info] 	published delta-kernel-api to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-api/0.1.0-SNAPSHOT/delta-kernel-api-0.1.0-SNAPSHOT-sources.jar
[info] 	published delta-kernel-api to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-api/0.1.0-SNAPSHOT/delta-kernel-api-0.1.0-SNAPSHOT-javadoc.jar
[info] 	published delta-kernel-default to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-default/0.1.0-SNAPSHOT/delta-kernel-default-0.1.0-SNAPSHOT.pom
[info] 	published delta-kernel-default to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-default/0.1.0-SNAPSHOT/delta-kernel-default-0.1.0-SNAPSHOT.jar
[info] 	published delta-kernel-default to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-default/0.1.0-SNAPSHOT/delta-kernel-default-0.1.0-SNAPSHOT-sources.jar
[info] 	published delta-kernel-default to file:/Users/allison.portis/.m2/repository/io/delta/delta-kernel-default/0.1.0-SNAPSHOT/delta-kernel-default-0.1.0-SNAPSHOT-javadoc.jar
```